lib/rubocop/cop/util.rb

```diff
@@ -187,7 +187,7 @@ def directions(side)
 
       def begins_its_line?(range)
         source_before_end = range.source_buffer.source[0...range.begin_pos]
-        source_before_end =~ /\n\s*\Z/
+        source_before_end.reverse!.sub!(/\A[ \t\r\f]*/, '').start_with?("\n")
```

Comment:
Yeah, actually if `rstrip` took an argument (as `String#split` does), that could probably have been used instead.

But as mentioned, first I need to get my benchmarking straight :-)